subsys/bluetooth/common/hci_common_internal.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Following build configurations use configurable CONFIG_BT_BUF_CMD_TX_COUNT:
+ * 1. Host + Controller build with and without Controller to Host data flow control, or
+ * 2. Host-only with and without Controller to Host data flow control, or
+ * 3. Controller-only without Controller to Host data flow control support
+ */
+#if !defined(CONFIG_BT_HCI_RAW) || !defined(CONFIG_HAS_BT_CTLR) || \
+	!defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+#if !defined(CONFIG_BT_HCI_RAW)
+
+/* Auto initiated additional HCI command buffers enqueued in the Host */
+
+#if defined(CONFIG_BT_REMOTE_VERSION)
+#define BT_BUF_CMD_TX_REMOTE_VERSION       1
+#else /* !CONFIG_BT_REMOTE_VERSION */
+#define BT_BUF_CMD_TX_REMOTE_VERSION       0
+#endif /* !CONFIG_BT_REMOTE_VERSION */
+
+#if defined(CONFIG_BT_AUTO_PHY_UPDATE)
+#define BT_BUF_CMD_TX_AUTO_PHY_UPDATE      1
+#else /* !CONFIG_BT_AUTO_PHY_UPDATE */
+#define BT_BUF_CMD_TX_AUTO_PHY_UPDATE      0
+#endif /* !CONFIG_BT_AUTO_PHY_UPDATE */
+
+#if defined(CONFIG_BT_AUTO_DATA_LEN_UPDATE)
+#define BT_BUF_CMD_TX_AUTO_DATA_LEN_UPDATE 1
+#else /* !CONFIG_BT_AUTO_DATA_LEN_UPDATE */
+#define BT_BUF_CMD_TX_AUTO_DATA_LEN_UPDATE 0
+#endif /* !CONFIG_BT_AUTO_DATA_LEN_UPDATE */
```

Comment:
I think some of these can be better replaced with `COND_CODE_1` to make it simpler:

e.g. 
```
#if defined(CONFIG_BT_REMOTE_VERSION)
#define BT_BUF_CMD_TX_REMOTE_VERSION       1
#else /* !CONFIG_BT_REMOTE_VERSION */
#define BT_BUF_CMD_TX_REMOTE_VERSION       0
#endif /* !CONFIG_BT_REMOTE_VERSION */
```

becomes 
```
#define BT_BUF_CMD_TX_REMOTE_VERSION COND_CODE_1(CONFIG_BT_REMOTE_VERSION, (1), (0))`
```

Reply:
Done